src/modules/navigator/mission.cpp

```diff
@@ -509,11 +509,15 @@ Mission::update_mission()
 	}
 
 	if (failed) {
+		// only warn if the check failed on merit
+		if ((int)_mission.count > 0) {
+			PX4_WARN("mission check failed");
```

Comment:
I find this confusing. I would try to move the `PX4_WARN` to the place where the actual check fails. And if the mission count is 0, it's a silent failure as you want it.